superset/assets/src/views/chartList/ChartList.tsx

```diff
@@ -0,0 +1,316 @@
+    SupersetClient.delete({
+      endpoint: `/api/v1/chart/${id}`,
+    }).then(
+      (resp) => {
+        const { lastFetchDataConfig } = this.state;
+        if (lastFetchDataConfig) {
+          this.fetchData(lastFetchDataConfig);
+        }
+        this.props.addSuccessToast(t('Deleted') + ` ${slice_name}`);
+      },
+      (err: any) => {
+        this.props.addDangerToast(t('There was an issue deleting') + `${slice_name}`);
+      },
+    );
+  }
+  public handleBulkDashboardDelete = (charts: Chart[]) => {
+    SupersetClient.delete({
+      endpoint: `/api/v1/dashboard/?q=!(${charts.map(({ id }) => id).join(',')})`,
+    }).then(
+      ({ json = {} }) => {
+        const { lastFetchDataConfig } = this.state;
+        if (lastFetchDataConfig) {
+          this.fetchData(lastFetchDataConfig);
+        }
+        this.props.addSuccessToast(json.message);
+      },
+      (err: any) => {
+        console.error(err);
+        this.props.addDangerToast(t('There was an issue deleting the selected dashboards'));
+      },
+    );
+  }
+
+  public fetchData = ({
+    pageIndex,
+    pageSize,
+    sortBy,
+    filters,
+  }: FetchDataConfig) => {
+    this.setState({ loading: true });
+    const filterExps = Object.keys(filters).map((fk) => ({
+      col: fk,
+      opr: filters[fk].filterId,
+      value: filters[fk].filterValue,
+    }));
+
+    const queryParams = JSON.stringify({
+      order_column: sortBy[0].id,
+      order_direction: sortBy[0].desc ? 'desc' : 'asc',
+      page: pageIndex,
+      page_size: pageSize,
+      ...(filterExps.length ? { filters: filterExps } : {}),
+    });
+
+    return SupersetClient.get({
+      endpoint: `/api/v1/chart/?q=${queryParams}`,
+    })
+      .then(({ json = {} }) => {
+        this.setState({ charts: json.result, chartCount: json.count, labelColumns: json.label_columns });
+      })
+      .catch(() => {
+        this.props.addDangerToast(
+          t('An error occurred while fetching Charts'),
+        );
+      })
+      .finally(() => {
+        this.setState({ loading: false });
+      });
+  }
+
+  public componentDidMount() {
+    SupersetClient.get({
+      endpoint: `/api/v1/chart/_info`,
+    })
+      .then(({ json = {} }) => {
+        this.setState({ filterTypes: json.filters, permissions: json.permissions });
+      });
+  }
+
+  public render() {
+    const { charts, chartCount, loading, filterTypes } = this.state;
+    return (
+      <div className='container welcome'>
+        <Panel>
+
+          <ConfirmStatusChange
+            title={t('Please confirm')}
+            description={t('Are you sure you want to delete the selected charts?')}
+            onConfirm={this.handleBulkDashboardDelete}
+          >
+            {(confirmDelete) => {
+              const bulkActions = [];
+              if (this.canDelete) {
+                bulkActions.push({
+                  key: 'delete',
+                  name: <><i className='fa fa-trash' /> Delete</>,
```

Comment:
Should we instead pass a`name` and (optional) `icon` and handle rendering tags on the other side?